uaclient/cli.py

```diff
@@ -1372,22 +1376,26 @@ def action_api(args, *, cfg):
 
 
 @assert_root
-@assert_lock_file("pro auto-attach")
-def action_auto_attach(args, *, cfg):
+def action_auto_attach(args, *, cfg: config.UAConfig) -> int:
     if cfg.is_attached:
```

Comment:
The reason to maintain this guard is backwards compatibility. I realize `_full_auto_attach` checks against this condition too but, it does not conditionally auto attach depending on `skip_auto_attach`. So, in order to continue informing the user that the reason to not auto attach is that 'the instance is already attached' with more preference than 'skip_auto_attach' we need to maintain this check guard here.

If the previous change is an acceptable change of behavior, we could simplify this by removing the `is_attached` check.

Reply:
This is a good point. I think we should also include checking for the `disable_auto_attach` feature on the API too. This will make that endpoint more strict and simplify the cli method. What do you think @orndorffgrant @renanrodrigo @CalvoM ?

Reply:
Yeah I think that makes sense, I can include that in the PR I'll put up touching the api full auto attach or we can do it in this PR

Reply:
draft pr that includes this here https://github.com/canonical/ubuntu-advantage-client/pull/2299

Reply:
Could there exists a situation where callers of the API want to auto attach but `UAConfig`'s `disable_auto_attach` is true?

As the public `full_auto_attach` doesn't expose the ability to pass in a different `UAConfig`, there would be no clean way to override that, but tightly-coupling clients to `UAConfig`/the actual file.

I am thinking about the `cloud-init` use-case, but this would apply to other clients.

Reply:
Per internal discussions, we think that this situation from `cloud-init`'s perspective is a corner case. Furthermore, if the consumer would really want to force the auto attach, then the config module `cc_write-files` can be used to modify the UA config file, prior `cc_ubuntu_advantage` execution.